src/FetchAuthTokenCache.php

```diff
@@ -188,4 +188,27 @@ public function getProjectId(callable $httpHandler = null)
 
         return $this->fetcher->getProjectId($httpHandler);
     }
+
+    /**
+     * Updates metadata with the authorization token.
+     *
+     * @param array $metadata metadata hashmap
+     * @param string $authUri optional auth uri
+     * @param callable $httpHandler callback which delivers psr7 request
+     * @return array updated metadata hashmap
+     */
+    public function updateMetadata(
```

Comment:
Should this end up calling the underlying implementation from the class it wraps? It looks like the implementations vary between credentials type.

Reply:
This is a great point. The reason I didn't do it this way is because I thought this would bypass the cache, as calling `parent::updateMetadata` would result in `parent::fetchAuthToken` method being called instead of `FetchAuthTokenCache::fetchAuthToken`. However, I think now that `parent::updateMetadata` would execute`FetchAuthTokenCache::fetchAccessToken` instead, and so making this a wrapper call is the way to go.
 We just need to ensure a test confirms this.

Reply:
After further consideration, this doesn't work because the Cache is a wrapper and not a subclass.

So we can:

1. Bypass the logic in `updateMetadata` - this is bad for obvious reasons
2. Bypass the cache in `fetchAuthToken` - this renders the class useless if `updateMetadata` is used
3. Cache the result of `fetchAuthToken` separately - this is somewhat useful, but still not optimal
4. Use the passed-in `$metadata` array to include the cached value, and update the existing classes so they only fill in the header if it isn't set - I don't think we want this behavior.
5. Leave `updateMetadata` out of the picture until v2 - this is the best option I see for now.